state/protocol/util/testing.go

```diff
@@ -28,3 +49,31 @@ func RunWithProtocolState(t testing.TB, f func(*badger.DB, *pbadger.State)) {
 		f(db, proto)
 	})
 }
+
+// use this function if you'd like to create your own protocol state with customized dependencies
+func RunWithProtocolStateDeps(t testing.TB, create createState, f func(*badger.DB, *pbadger.State)) {
+	unittest.RunWithBadgerDB(t, func(db *badger.DB) {
+		proto := WithProtocolState(t, db, create)
+		f(db, proto)
+	})
+}
+
+func RunWithProtocolStateAndConsumer(t testing.TB, consumer protocol.Consumer, f func(*badger.DB, *pbadger.State)) {
```

Comment:
Add a new method so that we can use a mock consumer for protcol state in tests, that mocks the `BlockProcessable` function